### PR TITLE
Update iml tags for assemblies.py

### PR DESCRIPTION
**Pull request by albeanth**
## What is the change?
Updates the impl tags to include additional information for an ongoing SQA effort. 

## Why is the change being made?
SQA fun. 

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- ~[Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify that the new/changed code works.~

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] If any [requirements](https://terrapower.github.io/armi/developer/tooling.html#watch-for-requirements) were affected, mention it in the [release notes](https://terrapower.github.io/armi/release/index.html).
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: armi/reactor/assemblies.py [hunk 1]
```diff
@@ -180,6 +180,14 @@ def add(self, obj: blocks.Block):
...
             :id: I_ARMI_ASSEM_BLOCKS
             :implements: R_ARMI_ASSEM_BLOCKS
+
+            Adds a unique Block to the top of the Assembly. If the Block already
+            exists in the Assembly, an error is raised in
+            :py:meth:`armi.reactor.composites.Composite.add`.
+            The spatialLocator of the Assembly is automatically updated
+            to account for the new Block. The spatial locator for the Block is
+            also updated in ``reestablishBlockOrder``. The axial mesh and other
+            block geometry parameters are updated in ``calculateZCoords``.
```
Comment (jyang-TP) [inline armi/reactor/assemblies.py:190]: We may want to add the following sentence before the sentence `The spatial locator for the Block is ...`

If blocks are axially mixed up, relocating each block with the proper axial mesh is performed in ``reestablishBlockOrder``.

**Review by jyang-TP — COMMENTED** [review]
The review has finished.

File: armi/reactor/assemblies.py (hunk 1 above)
Comment (albeanth) [reply armi/reactor/assemblies.py:190]: So the docstring for `reestablishBlockOrder` is misleading, in my opinion. That method actually doesn't do anything with the axial mesh of the assembly. It re-initializes the spatialGrid for the assembly (which confusingly, isn't used to calculate the axial mesh -- see Assembly.getAxialMesh()) and then updates the spatialLocator and name for the Block. 

I will update the impl tag description to reflect this.

Comment (albeanth) [reply armi/reactor/assemblies.py]: https://github.com/terrapower/armi/pull/1604/commits/73871c2bee583f47c4276bc0485f235e7c9e05fa

**Review by jyang-TP — APPROVED** [review]

**Pull request merged**